src/libstore/local-store.cc

```diff
@@ -1499,21 +1499,24 @@ bool LocalStore::verifyStore(bool checkContents, RepairFlag repair)
     auto fdGCLock = openGCLock();
     FdLock gcLock(fdGCLock.get(), ltRead, true, "waiting for the big garbage collector lock...");
 
-    StorePathSet store;
-    for (auto & i : readDirectory(realStoreDir)) {
-        try {
-            store.insert({i.name});
-        } catch (BadStorePath &) { }
-    }
+    StorePathSet validPaths;
 
-    /* Check whether all valid paths actually exist. */
-    printInfo("checking path existence...");
+    {
+        StorePathSet store;
```

Comment:
Since we're moving things around, maybe rename this, as `store` isn't really descriptive. Maybe something like `pathsInStoreDir`?